src/components/movies/MoviesIndex.js

```diff
@@ -1,3 +1,69 @@
+import { getAllMovies } from "../../api/fetch"
+import MovieListing from "./MovieListing"
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import ErrorMessage from "../errors/ErrorMessage";
+import "../shows/ShowsIndex.css";
+
+function filterMovies(search, movies) {
+  return movies.filter(movie => {
+    return movie.title.toLowerCase().includes(search.toLowerCase())
+  })
+}
+
 export default function MoviesIndex() {
-  return <p>Movie List</p>;
+  const [allMovies, setAllMovies] = useState([])
+  const [movies, setMovies] = useState([])
+  const [error, setError] = useState(false)
+  const [searchTitle, setSearchTitle] = useState("")
+
+ 
+  function handleTextChange(event){
+    setSearchTitle(event.target.value);
+     const filteredMovies = filterMovies(event.target.value, allMovies)
+     const result = event.target.value.length ? filteredMovies : allMovies 
+     setMovies(result);
+    
+  }
+
+  useEffect(() => {
+    getAllMovies().then(response => {
+      setAllMovies(response)
+      setMovies(response)
+      setError(false)
+    })
+    .catch((error) => {
+      console.log(error)
+      setError(true)
+    })
+  })
```

Comment:
You're missing the empty dependency here which is causing an endless loop of re-renders

Reply:
Thank you!

Reply:
Updated my code.